src/compiler/commandLineParser.ts

```diff
@@ -363,27 +369,31 @@ namespace ts {
         {
             name: "noUnusedLocals",
             type: "boolean",
+            affectsSemanticDiagnostics: true,
             showInSimplifiedHelpView: true,
             category: Diagnostics.Additional_Checks,
             description: Diagnostics.Report_errors_on_unused_locals,
         },
         {
             name: "noUnusedParameters",
             type: "boolean",
+            affectsSemanticDiagnostics: true,

```

Comment:
What determines whether affectsSemanticDiagnostics is true?

Reply:
The flags is true if changing value of the option, affects the semantic diagnostics reported(more/less) even if there is no change in file content. ( in turn whether to we need to throw away semantic diagnostics from cache(in --w mode) ) Eg. --noUnusedLocals can report more errors when turned on so it will have the flag true

Reply:
It seems easy to forget set this when adding a new flag. Would making it required help? It would be annoying then, though.

Reply:
yeah I don't think we want this required.